### ISSUE TITLE
Widget doesn't work in Mendix 7.22.1

### ISSUE THREAD

**Issue by sebasvanderplaat** [ivosturm/ArcGIS#2]
Hi, When we try to use the widget in Mendix 7.22.1 we get an error while bundling the widgets. Please see the attached log

[ArcGIS.log](https://github.com/ivosturm/ArcGIS/files/2745791/ArcGIS.log)

**Comment by sebasvanderplaat** [comment]
Hi Ivo,
We did some research and this is the answer to why the widget is not working in 7.22.1:

Heya, I did some more research. The bundling complains about some missing `dojox` dependencies and we did fix something there
If you look what `dojox` libraries we ship with the client (also in 7.19) you shee that this is a limited set of `dojox`
You can see that if you take a look in a Mendix installation in the directory `runtime\mxclientsystem\dojox`
So other stuff was not provided by the client at runtime
However when checking widget we use a separate set of file (located in `modeler\tools\grunt\dojox`) where there was the complete set of dojox
So the checker was able to find the depenencies, while the libs are not present during runtime which might result in runtime exceptions when you try to use the not shipped libraries during runtime
So we made this consistent, what we ship and what we check is now in sync

Embedding dojox in the custom widget and pointing the dojox dependencies should fix that

**Comment by ivosturm** [comment]
Hi Sebastiaan,

The Mendix dojox implementation was limited, hence I have added a set of dojox dependencies myself in the widget. Those files have already been included since the first release of the widget. For the rest it was depending on Mendix clientsystem.

Are you saying that some of those files that were part of Mendix clientsystem for dojox have been removed in some late Mendix 7.x release, hence I have to add to the widget dojox files as well?

Do you know which dependencies? From widget Mendix 7.x version does this start?

Kind regards,
Ivo

Van: sebasvanderplaat <notifications@github.com>
Verzonden: vrijdag 11 januari 2019 15:47
Aan: ivosturm/ArcGIS <ArcGIS@noreply.github.com>
CC: Subscribed <subscribed@noreply.github.com>
Onderwerp: Re: [ivosturm/ArcGIS] Widget doesn't work in Mendix 7.22.1 (#2)


Hi Ivo,
We did some research and this is the answer to why the widget is not working in 7.22.1:

Heya, I did some more research. The bundling complains about some missing dojox dependencies and we did fix something there
If you look what dojox libraries we ship with the client (also in 7.19) you shee that this is a limited set of dojox
You can see that if you take a look in a Mendix installation in the directory runtime\mxclientsystem\dojox
So other stuff was not provided by the client at runtime
However when checking widget we use a separate set of file (located in modeler\tools\grunt\dojox) where there was the complete set of dojox
So the checker was able to find the depenencies, while the libs are not present during runtime which might result in runtime exceptions when you try to use the not shipped libraries during runtime
So we made this consistent, what we ship and what we check is now in sync

Embedding dojox in the custom widget and pointing the dojox dependencies should fix that

**Comment by sebasvanderplaat** [comment]
Hi Ivo,

Thanks for the quick response. In 7.21 it still works, so the issue start from 7.22. For some reason Mendix decided to make the bundling check of the widgets more strict during a minor upgrade. I'm not exactly sure about what the dependencies are, but this action should fix it: _Embedding dojox in the custom widget and pointing the dojox dependencies should fix that_

**Comment by ivosturm** [comment]
Hi Sebastiaan,

For another client I am now also seeing this issue. We moved to 7.22.2. Did you already manage to solve it? I will be digging into it....

**Comment by sebasvanderplaat** [comment]
Hi Ivo,

We decided to go with the leaflet widget. We don’t have any updates on the development for the GIS widget.


Regards,


Sebastiaan

**Comment by ivosturm** [comment]
Hi Sebastiaan,

I just released a new version which is compatible with Mx 7.22 and up.

Can I ask why you went for the leaflet map widget? It does not have the same functionalities right? Or is it just because you couldn’t upgrade to Mx 7.22.x?

Let me know, thanks!

Kind regards,
Ivo

Van: sebasvanderplaat <notifications@github.com>
Verzonden: woensdag 30 januari 2019 16:34
Aan: ivosturm/ArcGIS <ArcGIS@noreply.github.com>
CC: Ivo Sturm <ivo.sturm@firstconsulting.nl>; Comment <comment@noreply.github.com>
Onderwerp: Re: [ivosturm/ArcGIS] Widget doesn't work in Mendix 7.22.1 (#2)

Hi Ivo,

We decided to go with the leaflet widget. We don’t have any updates on the development for the GIS widget.


Regards,


Sebastiaan

**Comment by sebasvanderplaat** [comment]
Hi Ivo,

In the first instance it was because we couldn’t upgrade. Then we started thinking and realized that the leaflet widget can easily be extended and has shorter support lines. Considering that, we chose to go with the leaflet widget.


Regards,


Sebastiaan

**Comment by ivosturm** [comment]
Hi Sebastiaan,

Thanks for your reply. Is it also easily extended to interact with objects in Mendix? So for instance only showing the subset of GIS objects that have a reference in the Mendix database? Or having an infowindow display both data from Mendix as well as ArcGIS? Or toggling visibility of layers to be shown on the map?

Those were the use cases for me to create the widget, so just out of curiosity. It does not make sense if I support a widget which has similar capabilities as the leaflet app :).

Thanks and your feedback is appreciated!

Regards,
Ivo

Op 30 jan. 2019 om 20:22 heeft sebasvanderplaat <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:

Hi Ivo,

In the first instance it was because we couldn’t upgrade. Then we started thinking and realized that the leaflet widget can easily be extended and has shorter support lines. Considering that, we chose to go with the leaflet widget.


Regards,


Sebastiaan
